az-digital/az_quickstart
# Configuration Management/Config Distro Integration

This pull request adds the `config_sync` / `config_distro` ecosystem to quickstart and creates a `config_provider` plugin that offers similar behavior to our existing system of handling overrides to contrib module configuration.

## Description
This pull request adds the `config_sync` / `config_distro` ecosystem to quickstart and creates a `config_provider` plugin that offers similar behavior to our existing system of handling overrides to contrib module configuration; this operates on a `config/quickstart` folder and has the effect of adding the configuration override change to whichever module originally owned the configuration.

For example, if the `cas` module originally deployed the `cas.settings` configuration, the additional `config_provider` plugin will make `az_cas`'s version of `config/quickstart/cas.settings.yml` visible to `config_sync` as an update to the `cas`modules settings, in other words allowing our overrides to participate in the `config_distro`ecosystem.

This creates the issue that a new install of quickstart has several pending distribution updates out of the gate (e.g., `cas` showing the updates from `az_cas`'s being enabled. This is resolved by doing an automatic import of the `config/quickstart` overrides when the module containing them is first enabled. This means, for example, that `az_cas`'s generated updated to `cas` config is automatically imported when `az_cas` is first enabled.

`config_sync` is D9 ready, but has one issue with the core version requirement on a mostly-stub testing submodule it contains. See my Drupal.org issue about this [here](https://www.drupal.org/project/config_sync/issues/3176955). This is a patch against an .info file, so I also have a gist [here](https://gist.github.com/tadean/602c6412fbbdcc16a2d8a0fb58642f9e) for the current packaged version that takes into account the drupal.org autopackaging additions.

A new unit test is added to verify that configuration overrides work correctly.

## Related Issue
#175 

## How Has This Been Tested?

- Verify that existing `az_cas` settings are imported.
- Make local changes to the embedded configuration of quickstart distribution modules, e.g. `az_flexible_page`, `az_paragraphs`, etc.
- Verify that the file changes are visible as distribution updates at `/admin/config/development/distro`

## Types of changes


## Checklist:
